iothub/device/src/InternalClient.cs

```diff
@@ -3,33 +3,34 @@
 
 using System;
 using System.Collections.Generic;
+using System.IO;
 using System.Linq;
+using System.Net.Sockets;
+using System.Net.WebSockets;
 using System.Threading;
 using System.Threading.Tasks;
+using DotNetty.Transport.Channels;
 using Microsoft.Azure.Devices.Client.Exceptions;
 using Microsoft.Azure.Devices.Client.Transport;
 using Microsoft.Azure.Devices.Client.Utilities;
 
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// Contains methods that a device can use to send messages to and receive messages from the service,
+    /// Contains methods that a client can use to send messages to and receive messages from the service,
     /// respond to direct method invocations from the service, and send and receive twin property updates.
     /// </summary>
-    internal class InternalClient : IDisposable
+    public class InternalClient : IDisposable
     {
+        private protected readonly PipelineContext _pipelineContext;
+        private protected readonly IotHubClientOptions _clientOptions;
```

Comment:
these are referenced by IotHubDeviceClient and IotHubModuleClient, so making them private protected.

Reply:
Maybe they should be properties with a protected getter and private setter?

Reply:
Sure, that'll work. I'll rework this with the change to make the device client/ module client only add to the context and not necessarily build it entirely.